src/k8s/pkg/k8sd/app/app.go

```diff
@@ -83,23 +91,36 @@ func New(cfg Config) (*App, error) {
 	}
 	app.readyWg.Add(1)
 
-	app.nodeConfigController = controllers.NewNodeConfigurationController(
-		cfg.Snap,
-		app.readyWg.Wait,
-	)
+	if !cfg.DisableNodeConfigController {
+		app.nodeConfigController = controllers.NewNodeConfigurationController(
+			cfg.Snap,
+			app.readyWg.Wait,
+		)
+	} else {
+		log.Println("node config controller was not initialized. change this behvaiour by unsetting --disable-node-config-controller (or setting it to `false` which is the default).")
```

Comment:
typo, but also unnecessarily verbose.

let's be [laconic](https://www.merriam-webster.com/dictionary/laconic)

```
log.Println("node-config-controller disabled via config")
```

same below